modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/components/Violations.tsx

```diff
@@ -44,122 +41,27 @@ const CATEGORY_FILTER_OPTIONS = [
 ];
 
-type ViolationsFilterProps = {
-	selectedCategories: Array<String>;
-	selectedImpact: Array<ImpactValue>;
-};
-
-function ViolationsFilter({
-	selectedCategories,
-	selectedImpact,
-}: ViolationsFilterProps) {
-	const [dropdownExpanded, setDropdownExpanded] = useState(false);
-	const dispatch = useFilteredViolationsDispatch();
-
-	return (
-		<>
-			<ClayDropDown
-				active={dropdownExpanded}
-				closeOnClickOutside
-				onActiveChange={setDropdownExpanded}
-				trigger={
-					<ClayButtonWithIcon
-						aria-label={Liferay.Language.get(
-							'open-violations-filter'
-						)}
-						displayType="unstyled"
-						small
-						symbol="filter"
-					/>
-				}
-			>
-				<ClayDropDown.ItemList>
-					<ClayDropDown.Group
-						header={Liferay.Language.get('filter-by-impact')}
-					>
-						{IMPACT_FILTER_OPTIONS.map((item) => {
-							const existsImpact =
-								selectedImpact.findIndex(
-									(element) => element === item.value
-								) !== -1;
-
-							return (
-								<ClayDropDown.Section
-									active={existsImpact}
-									key={item.value}
-								>
-									<ClayCheckbox
-										checked={existsImpact}
-										data-testid={item.value}
-										label={item.label}
-										onChange={() => {
-											dispatch({
-												payload: {
-													value: item.value,
-												},
-												type: existsImpact
-													? TYPES.IMPACT_REMOVE
-													: TYPES.IMPACT_ADD,
-											});
-										}}
-									></ClayCheckbox>
-								</ClayDropDown.Section>
-							);
-						})}
-					</ClayDropDown.Group>
-					<ClayDropDown.Group
-						header={Liferay.Language.get('filter-by-category')}
-					>
-						{CATEGORY_FILTER_OPTIONS.map((item) => {
-							const existsCategory =
-								selectedCategories.findIndex(
-									(element) => element === item.value
-								) !== -1;
-
-							return (
-								<ClayDropDown.Section key={item.value}>
-									<ClayCheckbox
-										checked={existsCategory}
-										data-testid={item.value}
-										label={item.label}
-										onChange={() => {
-											dispatch({
-												payload: {
-													value: item.value,
-												},
-												type: existsCategory
-													? TYPES.CATEGORY_REMOVE
-													: TYPES.CATEGORY_ADD,
-											});
-										}}
-									></ClayCheckbox>
-								</ClayDropDown.Section>
-							);
-						})}
-					</ClayDropDown.Group>
-				</ClayDropDown.ItemList>
-			</ClayDropDown>
-		</>
-	);
-}
-
 type TViolationNext = {
 	violationIndex: number;
 };
 
 type ViolationsProps = {
 	next?: (payload: TViolationNext) => void;
+	onFilterChange: (type: keyof typeof TYPES, value: string) => void;
 	selectedCategories: Array<String>;
 	selectedImpact: Array<ImpactValue>;
 	violations: Array<Result>;
 };
 
 export default function Violations({
 	next,
+	onFilterChange,
 	selectedCategories,
 	selectedImpact,
 	violations,
 }: ViolationsProps) {
+	const [expanded, setExpanded] = useState(false);
```

Comment:
I remember in Clay We had a pattern to type the state like `useState<boolean>(false)`. In this case, TS is inferring, right?

Reply:
Yeah in Clay we did that but then we stopped adding and let TypeScript do the inference in most use cases.